app/assets/javascripts/fae/form/inputs/_text.js

```diff
@@ -128,5 +164,71 @@ Fae.form.text = {
       },
       resetCss: true
     });
-  }
-};
+  },
+
+    /**
+   * Find all translate fields and initialize them with button
+   */
+    initTranslation: function () {
+      // var $translate_button = $('.js-translate-button');
+      $('.js-translate-button').click(function(e) {
+        var $this = $(this);
+        var translate_field;
+
+        // grab the field the button belongs to
+        if (this.closest(".text")) {
+          translate_field = this.closest(".text");
+        } else  {
+          translate_field = this.closest(".string");
+        }
+
+        // grab language and model name from field, container logic is for image alt text
+        var translate_language = translate_field.attributes['data-language'].value;
+        var translate_model
+        if (translate_field.className.includes("container")) {
+          translate_model = translate_field.className.split(' ').reverse()[1];
+        } else {
+          translate_model = translate_field.className.split(' ').pop();
+        }
+
+        // fix model name for static pages
+        var n = translate_model.lastIndexOf('content');
+        if (n) {
+          translate_model = translate_model.slice(0, n) + translate_model.slice(n).replace('content', 'attributes_content');
+        }
+        // fix model name for image alt text
+        var n = translate_model.lastIndexOf('_alt');
+        if (n) {
+          translate_model = translate_model.slice(0, n) + translate_model.slice(n).replace('_alt', '_attributes_alt');
+        }
+
+        // set english model name and use that to get text from english field
+        var english_model = translate_model.replace('_' + translate_language, '_en')
+        var english_text = $('#' + english_model)[0].value 
+
+        // get translate_language in correct format for request
+        if (translate_language.length == 4) {
+          translate_language = `${translate_language.slice(0,2)}-${translate_language.slice(2)}`
+        }
+
+        $.ajax({
+          url: Fae.path + '/translate_text',
+          type: "post",
+          data: { translation_text: { language: translate_language, en_text: english_text } },
+          success: function(data) {
```

Comment:
we should probably have a check to ensure the `data` being returned from the response exists and is in the format you expect before continuing on any further in the function. since your accessing it like `data[0]`, something like `if (data && data.length > 0)` or whatever seems appropriate

Reply:
updated!